src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/View/AbstractViewHandler.cs

```diff
@@ -43,16 +43,28 @@ protected AbstractViewHandler(PropertyMapper mapper)
 
 		public object? NativeView => TypedNativeView;
 
-		public virtual void SetView(IView view)
+
+		public void SetView(IView view)
 		{
 			_ = view ?? throw new ArgumentNullException(nameof(view));
 
+			if (VirtualView != null)
+			{
+				TearDownOldVirtualView(VirtualView);
+				_mapper.UpdateProperty(this, VirtualView, nameof(TearDownOldVirtualView));
```

Comment:
Is this the best way to hook this? I wanted a way for people to be able to wire into these life cycle points without having to override the handler itself